vm/src/obj/objbytes.rs

```diff
@@ -201,3 +203,58 @@ fn bytes_iter(obj: PyBytesRef, _vm: &VirtualMachine) -> PyIteratorValue {
         iterated_obj: obj.into_object(),
     }
 }
+
+fn compare_slice(a: &[u8], b: &[u8]) -> bool {
+    for (i, j) in a.iter().zip(b.iter()) {
+        if i != j {
+            return false;
+        }
+    }
+    return true;
+}
+
+fn compare_vec(a: &Vec<u8>, b: &Vec<u8>) -> bool {
+    let a_len = a.len();
+    let b_len = b.len();
+    for (n, i) in a.iter().enumerate() {
+        if n + b_len <= a_len && *i == b[0] {
+            if compare_slice(&a[n..n + b_len], b) {
+                return true;
+            }
+        }
+    }
+    false
+}
+
+impl PyBytesRef {
+    fn add(self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        if objtype::isinstance(&other, &vm.ctx.bytes_type()) {
+            let rhs = get_value(&other);
+            let elements: Vec<u8> = self.value.iter().chain(rhs.iter()).cloned().collect();
+            Ok(vm.ctx.new_bytes(elements))
+        } else {
+            Err(vm.new_not_implemented_error("".to_string()))
+        }
+    }
+
+    fn contains(self, needle: PyObjectRef, vm: &VirtualMachine) -> PyResult<bool> {
+        if objtype::isinstance(&needle, &vm.ctx.bytes_type()) {
+            if compare_vec(&self.value, &get_value(&needle)) {
+                return Ok(true);
+            } else {
+                return Ok(false);
+            }
```

Comment:
You could use here:
```rust
let result = compare_vec(&self.value, &get_value(&needle));
Ok(result)
```
Notice the lack of `if`-statement, and lack of `return` here.